### DISCUSSION TITLE
Conditional array validation without going deep into superRefine

### DISCUSSION THREAD

**Discussion by tonyjara** [colinhacks/zod#2211] [Q&A]
Hi, I have this:
```ts
const examplesSchema = z.object({
  status: z.string(),
  someOtherObject: z
    .object({
      field1: z.string().min(1, 'Minimum 1'),
      field2: z.string(),
      field3: z.string().min(13, 'Minimum 13'),
    })
    .array()
    .superRefine((val) => {
      // Access status here?
      return false;
    }),
});

```
Is there a way yo say if status is equal to a specific string make optional, if not validate? Otherwise I have to go into super refine and I'm not even sure how to do that for arrays. Thank you!

**Comment by JacobWeisenburger** [comment]
I'm not sure I understand, because if `status` is set to a specific string, then it would not be `undefined`, so making it optional would be pointless. Is there something I'm missing?

But if this is in fact what you are asking for, I think this should be close to what you want.
```ts
const examplesSchema = z.object( {
    status: z.literal( 'a specific string' ).optional().or( z.string() ),
    someOtherObject: z.object( {
        field1: z.string().min( 1, 'Minimum 1' ),
        field2: z.string(),
        field3: z.string().min( 13, 'Minimum 13' ),
    } ).array()
} )
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
